### PR TITLE
Cursed assembler

### PR DESCRIPTION
**Pull request by boubou19**
add compat for the cursed launcher

### CODE REVIEW

File: releases/manifests/nightly.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "version": "nightly",
-  "last_version": "2.1.2.0",
-  "last_updated": "2022-08-28T17:31:58.091279",
+  "last_version": null,
```
Comment (mitchej123) [inline releases/manifests/nightly.json:3]: this should be 2.1.2.0; was fixed in the branch you merged in

File: src/gtnh/modpack_manager.py [hunk 2]
```diff
@@ -719,6 +748,38 @@ def set_github_mod_side(self, mod_name: str, side: str) -> bool:
...
+            log.error(f"Release `{Fore.RED}{mod_name} is not an external mod{Fore.RESET}")
+            return False
+
+        if mod.side == side:
+            log.warn(f"{Fore.YELLOW}{mod.name}'s side is already set to {side}{Fore.RESET}")
+            return False
+
+        if side in [s.name for s in Side]:
+            mod.side = Side[side]
+            # idk a better way of doing this
```
Comment (mitchej123) [inline src/gtnh/modpack_manager.py:764]: You shouldn't have to do this at all.  The `get_external_mod` should return the actual object, so by updating the mod, you've updated the mod held in the list and the dictionary.

File: gtnh-modpack.json [hunk 3]
```diff
@@ -1,6 +1,7 @@
 {
   "releases": [
     "nightly",
+    "2.1.2.4-pre29",
```
[the remark this answers is not in the document]
Comment (boubou19) [reply gtnh-modpack.json:4]: in the gui they should be sorted by last modified index.

File: src/gtnh/modpack_manager.py (hunk 2 above)
Comment (boubou19) [reply src/gtnh/modpack_manager.py:764]: ah i see, got too cautious about outdated cache

**Review by mitchej123 — APPROVED** [review]

**Pull request merged**